docs/serverdev/index.rst

```diff
@@ -0,0 +1,130 @@
+and their libraries on OSX.
+
+Python
+******
+While OSX ships with an acceptable version of Python, installing the latest binaries from HomeBrew
+is recommended. In addition, it is highly recommended to separate your libraries per-environment
+with the use of virtual environments. One of the easiest ways to curate your virtual environments is
+with `virtualenvwrapper <https://virtualenvwrapper.readthedocs.org/>`_. Run the following to install
+and configure brewed Python and virtualenvwrapper::
+
+    brew install python
+    pip install virtualenvwrapper
+    touch ~/.bashrc >> source /usr/local/bin/virtualenvwrapper.sh
+    source ~/.bashrc
+    mkvirtualenv sentry
+
+JavaScript
+**********
+JavaScript is a tad easier because it separates environments already. To install node.js and
+the Node Package Manager(npm) run::
+
+    brew install nodejs
+
+Installing Libraries
+********************
+Once all of your dependencies are installed, head over to the root of ``getsentry/sentry``.
+Run the following to install both the Python and JavaScript libraries that Sentry depends on
+and some extra pieces that hold the development environment together::
+
+    make develop
+
+Running the Development Server
+------------------------------
+Before you are able to run the development server, you first must create a proper database
+for it to use. Running the following will create the proper database and fill it with example
+data::
+
+    createdb -E utf-8 sentry
+    sentry init
+    sentry upgrade
+    ./bin/load-mocks
+
+.. note:: You will be prompted to create a user during ``sentry upgrade``. It is recommended
+  to supply the prompts with a proper email address and password. It is also recommended to
+  designate said user as a superuser.
+
+Once you've successfully stood up your dataset, you can now run the development server::
+
+    sentry devserver
+
+.. note:: If you would like to be able to run ``devserver`` outside of your root checkout,
+  you can install ``webpack`` globally with ``npm install -g webpack``.
+
+When webpack finishes processing, you can find a login prompt for the user account you previously
+created at `<http://localhost:8000>`_. From there, you are free to browse the website as an
+administrator.
+
+Staging Your Changes
+--------------------
+You've made your changes to the codebase, now it's time to present them to the Sentry developers.
+It is recommended to first run the test suite locally in order to find any linting, syntax, or
+integration before you post a Pull Request.
+
+Code Coverage
+'''''''''''''
+#TODO Should this even be here?#
+
+Running the Test Suite Locally
+''''''''''''''''''''''''''''''
+There are no additional services required for running the Sentry test suite. To install dependent
+libraries, lint all source code, and run both the Python and JavaScript test suites, simply run::
+
+    make test
+
+.. note:: If you find yourself constantly running ``make test`` and wishing it was faster, check
```

Comment:
We could elaborate on this a bit and explain running `py.test` since that's what anyone who is working regularly uses.